src/runtime/tests/manifest-test.ts

```diff
@@ -3659,6 +3659,87 @@ resource SomeName
       `), `Type variable ~a does not contain field 'foo'`);
     });
 
+    it('fails to parse concrete types with inline schema field of `*`', async () => {
+      await assertThrowsAsync(async () => await parseManifest(`
+          particle Foo
+            data: reads {*}
+      `));
+      await assertThrowsAsync(async () => await parseManifest(`
+          particle Foo
+            data: reads {name: Text, *}
+      `));
+    });
+
+    it('warns about using multiple `*` in a single variable constraint', async () => {
+      const manifest = await parseManifest(`
+          particle Foo
+            data: reads ~a with {*, *}
+      `);
+
+      assert.lengthOf(manifest.errors, 1);
+      assert.equal(manifest.errors[0].key, 'multiStarFields');
+    });
+
+    it('supports field-level checks and claims with max type variables', async () => {
```

Comment:
You seem to be testing a lot of things in this unit tests. I am pondering whether we could / should break them down.

We could split this into:
1) Ensuring checks and claims can be parsed and are represented in data structured properly. (here in manifest-test.ts)
2) Ensuring type variables with `*` pseudo fields are in type.ts data structure properly. (adding assertion to the tests you have above, beyond checking that the manifest parses).
3) The test for type inference when using "max type variables" could be in `type-checker-test.ts`. I know we have a similar test there already, but you could add one (or update existing) to use Manifest.parse()
4) The test for ensuring that checks and claims are actually operations with type variables and maxTypeVars in particular could be placed among other DFA tests.

Reply:
May I save (4) for a future PR? I'm made a pass at the rest.

Reply:
Sure 👍

Reply:
I don't see (1).

i.e. I would like a test where you parse:

```
      particle Foo
        data: reads ~a with {age: Number, *}
```

And you have an assert that the type variable in the connection type has the right constraint and the typevarinfo has resolveToMax set to true.